### DISCUSSION TITLE
OS specific sections?

### DISCUSSION THREAD

**Discussion by fkromer** [rust-embedded/book#376] [Q&A] [answered]
At what types of embedded devices (from an OS point of view) is this repo targeting at? Shall we add OS specific sections?

Background:
In my opinion a distinction between low end devices running a RTOS (Zephyr RTOS, FreeRTOS, …), a „super loop“ and high end devices running a embedded Linux distro (Ubuntu Core, Debian, Yocto Linux, …) makes a lot of sense. The reason is quite different ecosystems (e.g. hardware abstraction and build considerations).

**Comment by BartMassey** [comment, marked as answer]
I think the Discovery Book (whether this book or <https://github.com/rust-embedded/discovery-mb2> ) is primarily targeted at the low-level experience: "super loop" or even lower. The new book will probably get a chapter on RTIC and Embassy sometime soon, but honestly there's enough to do to teach the unique aspects of working with hardware to keep a reader busy for a long time.

**Comment by fkromer** [comment]
Do you have some reference to the new book? I was wondering where really interesting things like an ISO 26262 ASIL D Qualified compiler for the Infineon AURIX platform (OS: RTOS) would fit into. This example is platform instead of OS specific.

https://hightec-rt.com/rust

**Reply by BartMassey** [reply]
That link above to <https://github.com/rust-embedded/discovery-mb2> (oops, was borked, fixed) is to the repo for the work-in-progress new edition of the Discovery Book. The published version is at <https://docs.rust-embedded.org/discovery-mb2/>.

I think the kind of stuff you're describing would be awesome in some to-be-created book; we should really consider the creation of a "Beyond Discovery" book for embedded. I think we're already running into limitations of size and scope for the Disco Book itself, though, which is really intended as a friendly introduction to embedded for beginners.

**Reply by fkromer** [reply]
I‘d be really interested in contributing to some bigger scope effort.

I‘ve started an effort to help Python programmers to use Rust as part of their Python applications https://github.com/fkromer/r4py. It uses https://squidfunk.github.io/mkdocs-material/ which is great for interactive code examples (code line highlighting, code annotations, …). Probably that’s something you want to consider as well for further efforts.
